### PR TITLE
Bug 1932453: Update format for Japanese date/times

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Thanks @kyoto for pointing out an issue with the format of the Japanese dates/times.

I refactored our dates/times to rely on Moment "LLL" locale defaults so we don't need to worry about syntax. I overrode the "LLL" default in Moment for English so we could retain our existing date/time format (no year). This should also fix any other issues with our existing languages and prevent issues with any additional languages we add.

English example (with override):
Screen Shot 2021-02-24 at 2 07 29 PM

Japanese example (with defaults):
Screen Shot 2021-02-24 at 2 08 06 PM

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1932453.

### CODE REVIEW

**Comment by rebeccaalpert** [comment]
Need to fix the timestamp tooltips. Putting on hold temporarily.

**Comment by rebeccaalpert** [comment]
I think we'll save ourselves a lot of grief if we just move this approach over to Moment and get rid of the i18n markup for dates. It looks like we can override the longDateFormat for "LLL", etc. to make them similar to the existing timestamp formatting. It looks like it works for Japanese and English; I'm just doing some cleanup and then I'll update the PR.

**Comment by rebeccaalpert** [comment]
Have to fix some tests; moving this into WIP status for now. I'll finish it tomorrow.

**Comment by kyoto** [comment]
Thanks @rebeccaalpert. This approach LGTM.

**Comment by rebeccaalpert** [comment]
```
displays Couchbase Operator CouchbaseCluster creation form: Interacting with a single namespace install mode Operator (Couchbase Operator) displays Couchbase Operator CouchbaseCluster creation form expand_less30sAssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test-id="dropdown-button"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13639:57) | displays Couchbase Operator CouchbaseCluster creation form: Interacting with a single namespace install mode Operator (Couchbase Operator) displays Couchbase Operator CouchbaseCluster creation form expand_less | 30s | AssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test-id="dropdown-button"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13639:57)
-- | -- | -- | --
displays Couchbase Operator CouchbaseCluster creation form: Interacting with a single namespace install mode Operator (Couchbase Operator) displays Couchbase Operator CouchbaseCluster creation form expand_less | 30s
AssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test-id="dropdown-button"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13639:57)
creates a Couchbase Operator CouchbaseCluster instance via the form: Interacting with a single namespace install mode Operator (Couchbase Operator) creates a Couchbase Operator CouchbaseCluster instance via the form expand_less30sAssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test="create-dynamic-form"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13635:8) | creates a Couchbase Operator CouchbaseCluster instance via the form: Interacting with a single namespace install mode Operator (Couchbase Operator) creates a Couchbase Operator CouchbaseCluster instance via the form expand_less | 30s | AssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test="create-dynamic-form"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13635:8)
creates a Couchbase Operator CouchbaseCluster instance via the form: Interacting with a single namespace install mode Operator (Couchbase Operator) creates a Couchbase Operator CouchbaseCluster instance via the form expand_less | 30s
AssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test="create-dynamic-form"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13635:8)
displays details about Couchbase Operator CouchbaseCluster instance on the "Details" tab: Interacting with a single namespace install mode Operator (Couchbase Operator) displays details about Couchbase Operator CouchbaseCluster instance on the "Details" tab expand_less30sAssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test-operand-link="cb-example"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13646:60) | displays details about Couchbase Operator CouchbaseCluster instance on the "Details" tab: Interacting with a single namespace install mode Operator (Couchbase Operator) displays details about Couchbase Operator CouchbaseCluster instance on the "Details" tab expand_less | 30s | AssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test-operand-link="cb-example"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13646:60)
displays details about Couchbase Operator CouchbaseCluster instance on the "Details" tab: Interacting with a single namespace install mode Operator (Couchbase Operator) displays details about Couchbase Operator CouchbaseCluster instance on the "Details" tab expand_less | 30s
AssertionError: Timed out retrying after 30000ms: Expected to find element: `[data-test-operand-link="cb-example"]`, but never found it.     at Context.eval (https://console-openshift-console.apps.ci-op-8nv9ldgp-75d12.origin-ci-int-gce.dev.openshift.com/__cypress/tests?p=tests/operator-install-single-namespace.spec.ts:13646:60)
```

I think this is a flake; I'm going to go ahead and take this out of WIP status.

/retest

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
Holding until #8289 merges to unblock CI.

/hold

**Pull request merged**
